ui/app/src/components/PreviewBrowseComponentsPanel/PreviewBrowseComponentsPanel.tsx

```diff
@@ -164,15 +141,23 @@ export function PreviewBrowseComponentsPanel() {
             </Select>
           )}
         </div>
-        <React.Suspense fallback={<LoadingState title={formatMessage(translations.loading)} />}>
+        {(!componentsState.isFetching &&
+          nnou(componentsState.pageNumber) &&
+          nnou(componentsState.page[componentsState.pageNumber])) ||
+        !componentsState.contentTypeFilter ? (
           <PreviewBrowseComponentsPanelUI
-            componentsResource={resource}
+            items={items}
+            count={componentsState.count}
+            pageNumber={componentsState.pageNumber}
+            limit={componentsState.query.limit}
             onPageChanged={onPageChanged}
             onRowsPerPageChange={onRowsPerPageChange}
             onDragStart={onDragStart}
             onDragEnd={onDragEnd}
           />
-        </React.Suspense>
+        ) : (
+          <LoadingState title={formatMessage(translations.loading)} />
+        )}
```

Comment:
As discussed in other occasions, show loading only when fetching is being done, not when there's nothing else to display.